apps/frontend/src/libs/components/checkbox/checkbox.tsx

```diff
@@ -1,73 +1,131 @@
+import { ZERO_INDEX } from "~/libs/constants/constants.js";
+import { getValidClassNames } from "~/libs/helpers/helpers.js";
 import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 import {
-	type Control,
-	type FieldPath,
 	type FieldValues,
+	type FormFieldProperties,
 	type InputOption,
 } from "~/libs/types/types.js";
 
 import styles from "./styles.module.css";
 
 type Properties<T extends FieldValues> = {
-	control: Control<T, null>;
+	hasVisuallyHiddenLabel?: boolean;
 	label: string;
-	name: FieldPath<T>;
+	onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
 	options: InputOption[];
-};
+	variant: "general" | "rounded";
+} & FormFieldProperties<T>;
 
 const Checkbox = <T extends FieldValues>({
 	control,
+	hasVisuallyHiddenLabel = false,
 	label,
 	name,
+	onChange,
 	options,
+	variant,
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
-	const { onChange, value } = field;
+	const { onChange: onFieldChange, value } = field;
+	const isGeneralCheckbox = variant === "general";
 
 	const handleCheckboxesChange = useCallback(
 		(event: React.ChangeEvent<HTMLInputElement>): void => {
 			const isChecked = event.target.checked;
-			const inputValue = Number(event.target.value);
+			const rawInputValue = event.target.value;
+			const inputValue = Number.isNaN(rawInputValue)
+				? rawInputValue
+				: Number(rawInputValue);
```

Comment:
If the value is string, then `Number.isNaN` will return false. We don't need this logic. Maybe you can try to make `InputOption` type generic and go from there